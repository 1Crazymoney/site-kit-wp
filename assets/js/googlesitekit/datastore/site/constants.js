export const STORE_NAME = 'core/site';
