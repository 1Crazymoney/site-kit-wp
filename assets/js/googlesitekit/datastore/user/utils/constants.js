export const INVALID_DATE_INSTANCE_ERROR = 'Date param must construct to a valid date instance or be a valid date instance itself.';
export const INVALID_DATE_STRING_ERROR = 'Invalid dateString parameter, it must be a string.';
