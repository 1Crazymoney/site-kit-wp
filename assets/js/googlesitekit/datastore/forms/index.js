/**
 * core/forms Data store: values
 *
 * Site Kit by Google, Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import Data from 'googlesitekit-data';
import invariant from 'invariant';
import { STORE_NAME } from './constants';

export { STORE_NAME };

const SET_FORM_VALUES = 'SET_FORM_VALUES';

export const INITIAL_STATE = {};

export const actions = {
	setValues( formName, formData ) {
		invariant( formName, 'formName is required for setting values.' );

		return {
			payload: { formName, formData },
			type: SET_FORM_VALUES,
		};
	},
};

export const controls = {};

export const reducer = ( state, { type, payload } ) => {
	switch ( type ) {
		case SET_FORM_VALUES: {
			const { formName, formData } = payload;

			return {
				...state,
				[ formName ]: formData,
			};
		}

		default: {
			return { ...state };
		}
	}
};

export const resolvers = {};

export const selectors = {
	getValue( state, formName, key ) {
		return state[ formName ][ key ];
	},
};

const store = {
	INITIAL_STATE,
	actions,
	controls,
	reducer,
	resolvers,
	selectors,
};

// Register this store on the global registry.
Data.registerStore( STORE_NAME, store );

export default store;
