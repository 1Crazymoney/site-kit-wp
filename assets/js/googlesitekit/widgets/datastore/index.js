/**
 * core/widgets data store
 *
 * Site Kit by Google, Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * Internal dependencies
 */
import Data from 'googlesitekit-data';
import areas from './areas';
import widgets from './widgets';

export const INITIAL_STATE = Data.collectState(
	areas.INITIAL_STATE,
	widgets.INITIAL_STATE,
);

import { STORE_NAME } from './constants';
export { STORE_NAME };

export const actions = Data.addInitializeAction(
	Data.collectActions(
		Data.commonActions,
		areas.actions,
		widgets.actions,
	)
);

export const controls = Data.collectControls(
	Data.commonControls,
	areas.controls,
	widgets.controls,
);

export const reducer = Data.addInitializeReducer(
	INITIAL_STATE,
	Data.collectReducers(
		areas.reducer,
		widgets.reducer,
	)
);

export const resolvers = Data.collectResolvers(
	areas.resolvers,
	widgets.resolvers,
);

export const selectors = Data.collectSelectors(
	areas.selectors,
	widgets.selectors,
);

const store = {
	actions,
	controls,
	reducer,
	resolvers,
	selectors,
};

// Register this store on the global registry.
Data.registerStore( STORE_NAME, store );

export default store;
