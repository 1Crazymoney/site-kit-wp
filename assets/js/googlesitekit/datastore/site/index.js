/**
 * External dependencies
 */

/**
 * Internal dependencies
 */
import { collect, collectReducers } from 'assets/js/googlesitekit/data/utils';
import connection from './connection';
import reset from './reset';

export const INITIAL_STATE = {
	connection: null,
	isFetchingConnection: false,
	isResetting: false,
};

export const STORE_NAME = 'core/site';

export const actions = collect(
	connection.actions,
	reset.actions,
);

export const controls = collect(
	connection.controls,
	reset.controls,
);

export const reducer = collectReducers( INITIAL_STATE, [
	connection.reducer,
	reset.reducer,
] );

export const resolvers = collect(
	connection.resolvers,
	reset.resolvers,
);

export const selectors = collect(
	connection.selectors,
	reset.selectors,
);

const store = {
	actions,
	controls,
	reducer,
	resolvers,
	selectors,
};

export const registerOn = ( registry ) => {
	registry.registerStore( STORE_NAME, store );
};

export default store;
