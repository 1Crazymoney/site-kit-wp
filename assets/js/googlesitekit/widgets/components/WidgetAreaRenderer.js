/**
 * WidgetAreaRenderer component.
 *
 * Site Kit by Google, Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import classnames from 'classnames';
import { string } from 'prop-types';
import useDynamicRefs from 'use-dynamic-refs';

/**
 * Internal dependencies
 */
import Data from 'googlesitekit-data';
import { STORE_NAME, WIDTH_GRID_MAP, WIDGET_WIDTHS } from '../datastore/constants';
import WidgetRenderer from './WidgetRenderer';

const { useSelect } = Data;

const resizeClasses = ( classNames, counter ) => {
	[ ...classNames ].reverse().some( ( _classNames, index ) => {
		const originalIndex = classNames.length - 1 - index;

		// Skip any classNames that are `null`; this happens when the component itself
		// renders `null`.
		if ( ! _classNames || ! Array.isArray( _classNames ) ) {
			return false;
		}

		if ( _classNames.includes( 'mdc-layout-grid__cell--span-3-desktop' ) ) {
			// Replace the 3-column class with a 4-column class so this element goes from 1/4
			// to 1/3 on desktop.
			classNames[ originalIndex ][ _classNames.indexOf( 'mdc-layout-grid__cell--span-3-desktop' ) ] = 'mdc-layout-grid__cell--span-4-desktop';

			counter -= 3;
		}

		if ( _classNames.includes( 'mdc-layout-grid__cell--span-6-desktop' ) ) {
			// Replace the 6-column class with a 8-column class so this element goes from 1/2
			// to 2/3 on desktop.
			classNames[ originalIndex ][ _classNames.indexOf( 'mdc-layout-grid__cell--span-6-desktop' ) ] = 'mdc-layout-grid__cell--span-8-desktop';

			counter -= 6;
		}

		return counter === 0;
	} );

	return [ classNames, counter ];
};

const WidgetAreaRenderer = ( { slug } ) => {
	const widgetArea = useSelect( ( select ) => select( STORE_NAME ).getWidgetArea( slug ) );
	const widgets = useSelect( ( select ) => select( STORE_NAME ).getWidgets( slug ) );

	const [ getRef, setRef ] = useDynamicRefs();

	let widgetClassNames = [];
	let counter = 0;
	widgets.forEach( ( widget, i ) => {
		widgetClassNames[ i ] = [];

		const widgetOutput = (
			<WidgetRenderer key={ widget.slug } slug={ widget.slug } />
		);

		// If this widget output `null`, there's no sense in outputting classes for it.
		if ( ! getRef( widget.slug )?.current ) {
			widgetClassNames[ i ] = null;
			return widgetOutput;
		}

		const width = widget.width;
		const classNamesForWidget = [ 'mdc-layout-grid__cell' ];

		if ( width === WIDGET_WIDTHS.FULL ) {
			classNamesForWidget.push(
				'mdc-layout-grid__cell--span-12',
			);
		}

		if ( width === WIDGET_WIDTHS.HALF ) {
			classNamesForWidget.push(
				'mdc-layout-grid__cell--span-6-desktop',
				'mdc-layout-grid__cell--span-8-tablet',
			);
		}

		if ( width === WIDGET_WIDTHS.QUARTER ) {
			classNamesForWidget.push(
				'mdc-layout-grid__cell--span-3-desktop',
				'mdc-layout-grid__cell--span-4-tablet',
			);
		}

		widgetClassNames[ i ] = classNamesForWidget;

		counter += WIDTH_GRID_MAP[ width ];

		if ( counter % 12 === 0 ) {
			counter = 0;
		}

		if ( counter > 12 ) {
			counter -= WIDTH_GRID_MAP[ width ];

			if ( counter === 9 ) {
				[ widgetClassNames, counter ] = resizeClasses( widgetClassNames, counter );
			}

			counter = WIDTH_GRID_MAP[ width ];
		}
	} );

	if ( counter === 9 ) {
		[ widgetClassNames, counter ] = resizeClasses( widgetClassNames, counter );
	}

	return (
		<div className={ classnames( 'mdc-layout-grid', 'googlesitekit-widget-area', `googlesitekit-widget-area--${ widgetArea.slug }`, `googlesitekit-widget-area--${ widgetArea.style }` ) }>
			<div className="mdc-layout-grid__inner">
				<header className={ classnames(
					'googlesitekit-widget-area-header',
					'mdc-layout-grid__cell--span-12-desktop'
				) }>
					<img alt="" src={ widgetArea.icon } />
					{ widgetArea.title &&
					<h3 className={ classnames(
						'googlesitekit-heading-3',
						'googlesitekit-widget-area-header__title'
					) }>
						{ widgetArea.title }
					</h3> }
					{ widgetArea.subtitle &&
					<h4 className="googlesitekit-widget-area-header__subtitle">{ widgetArea.subtitle }</h4>
					}
				</header>
			</div>
			<div className="googlesitekit-widget-area-widgets">
				<div className="mdc-layout-grid__inner">
					{ widgets.map( ( widget, i ) => {
						return (
							<WidgetRenderer
								className={ widgetClassNames[ i ] !== null ? classnames( widgetClassNames[ i ] ) : 'googlesitekit-widget-area--hidden' }
								key={ widget.slug }
								ref={ setRef( widget.slug ) }
								slug={ widget.slug }
							/>
						);
					} ) }
				</div>
			</div>
		</div>
	);
};

WidgetAreaRenderer.propTypes = {
	slug: string.isRequired,
};

export default WidgetAreaRenderer;
