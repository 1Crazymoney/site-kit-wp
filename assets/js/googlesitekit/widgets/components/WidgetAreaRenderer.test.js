/**
 * WidgetAreaRenderer component tests.
 *
 * Site Kit by Google, Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * WordPress dependencies
 */
import { forwardRef } from '@wordpress/element';

/**
 * Internal dependencies
 */
import WidgetAreaRenderer from './WidgetAreaRenderer';
import { STORE_NAME, WIDGET_WIDTHS } from '../datastore/constants';
import {
	createTestRegistry,
	muteConsole,
	render,
	unsubscribeFromAll,
} from '../../../../../tests/js/test-utils';

const createTestRegistryWithArea = ( name = 'gridcell-test' ) => {
	const registry = createTestRegistry();

	registry.dispatch( STORE_NAME ).registerWidgetArea( name, {
		title: 'Dashboard Header',
		subtitle: 'Cool stuff for yoursite.com',
		style: 'boxes',
	} );
	registry.dispatch( STORE_NAME ).assignWidgetArea( name, 'dashboard' );

	return registry;
};

const WidgetComponent = forwardRef( ( _props, ref ) => {
	return ( <div ref={ ref }>Foo bar!</div> );
} );

const WidgetComponentEmpty = () => {
	return null;
};

const createWidgets = ( registry, widgets ) => {
	widgets.forEach( ( { component, slug, width } ) => {
		registry.dispatch( STORE_NAME ).registerWidget( slug, {
			component,
			width,
		} );
		registry.dispatch( STORE_NAME ).assignWidget( slug, 'gridcell-test' );
	} );
};

describe( 'WidgetAreaRenderer', () => {
	let registry;

	beforeEach( () => {
		registry = createTestRegistryWithArea();
	} );

	afterEach( () => {
		unsubscribeFromAll( registry );
	} );

	it( 'should return the same number of elements as widgets from a selector', async () => {
		createWidgets( registry, [
			{ component: WidgetComponent, slug: 'one', width: WIDGET_WIDTHS.FULL },
			{ component: WidgetComponent, slug: 'two', width: WIDGET_WIDTHS.FULL },
			{ component: WidgetComponent, slug: 'three', width: WIDGET_WIDTHS.FULL },
		] );

		const widgets = registry.select( STORE_NAME ).getWidgets( 'gridcell-test' );
		expect( widgets ).toHaveLength( 3 );

		const { container } = render( <WidgetAreaRenderer slug="gridcell-test" />, { registry } );

		// expect( Object.values( container.firstChild.querySelectorAll('.googlesitekit-widget-area-widgets')[0].classList ) ).toEqual( [] );
		expect( container.firstChild.querySelectorAll( '.googlesitekit-widget-area-widgets' )[ 0 ] ).toMatchSnapshot();
	} );

	it( 'should resize widgets so they fill a row', () => {
		registry = createTestRegistryWithArea( 'gridcell-test' );
		createWidgets( registry, [
			{ component: WidgetComponent, slug: 'one', width: WIDGET_WIDTHS.FULL },
			{ component: WidgetComponent, slug: 'two', width: WIDGET_WIDTHS.QUARTER },
			{ component: WidgetComponent, slug: 'three', width: WIDGET_WIDTHS.HALF },
		] );

		let container = render( <WidgetAreaRenderer slug="gridcell-test" />, { registry } ).container;
		expect( container.firstChild.querySelectorAll( '.googlesitekit-widget-area-widgets' )[ 0 ] ).toMatchSnapshot();

		registry = createTestRegistryWithArea( 'gridcell-test' );
		createWidgets( registry, [
			{ component: WidgetComponent, slug: 'one', width: WIDGET_WIDTHS.QUARTER },
			{ component: WidgetComponent, slug: 'two', width: WIDGET_WIDTHS.FULL },
			{ component: WidgetComponent, slug: 'three', width: WIDGET_WIDTHS.QUARTER },
			{ component: WidgetComponent, slug: 'four', width: WIDGET_WIDTHS.QUARTER },
		] );
		container = render( <WidgetAreaRenderer slug="gridcell-test" />, { registry } ).container;
		expect( container.firstChild.querySelectorAll( '.googlesitekit-widget-area-widgets' )[ 0 ] ).toMatchSnapshot();

		registry = createTestRegistryWithArea( 'gridcell-test' );
		createWidgets( registry, [
			{ component: WidgetComponent, slug: 'one', width: WIDGET_WIDTHS.QUARTER },
			{ component: WidgetComponent, slug: 'two', width: WIDGET_WIDTHS.FULL },
			{ component: WidgetComponent, slug: 'three', width: WIDGET_WIDTHS.QUARTER },
			{ component: WidgetComponent, slug: 'four', width: WIDGET_WIDTHS.HALF },
			{ component: WidgetComponent, slug: 'five', width: WIDGET_WIDTHS.HALF },
		] );
		container = render( <WidgetAreaRenderer slug="gridcell-test" />, { registry } ).container;
		expect( container.firstChild.querySelectorAll( '.googlesitekit-widget-area-widgets' )[ 0 ] ).toMatchSnapshot();

		registry = createTestRegistryWithArea( 'gridcell-test' );
		createWidgets( registry, [
			{ component: WidgetComponent, slug: 'one', width: WIDGET_WIDTHS.QUARTER },
			{ component: WidgetComponent, slug: 'two', width: WIDGET_WIDTHS.QUARTER },
			{ component: WidgetComponent, slug: 'three', width: WIDGET_WIDTHS.HALF },
		] );
		container = render( <WidgetAreaRenderer slug="gridcell-test" />, { registry } ).container;
		expect( container.firstChild.querySelectorAll( '.googlesitekit-widget-area-widgets' )[ 0 ] ).toMatchSnapshot();

		registry = createTestRegistryWithArea( 'gridcell-test' );
		createWidgets( registry, [
			{ component: WidgetComponent, slug: 'one', width: WIDGET_WIDTHS.QUARTER },
			{ component: WidgetComponent, slug: 'two', width: WIDGET_WIDTHS.QUARTER },
			{ component: WidgetComponent, slug: 'three', width: WIDGET_WIDTHS.QUARTER },
			{ component: WidgetComponent, slug: 'four', width: WIDGET_WIDTHS.HALF },
		] );
		container = render( <WidgetAreaRenderer slug="gridcell-test" />, { registry } ).container;
		expect( container.firstChild.querySelectorAll( '.googlesitekit-widget-area-widgets' )[ 0 ] ).toMatchSnapshot();

		registry = createTestRegistryWithArea( 'gridcell-test' );
		createWidgets( registry, [
			{ component: WidgetComponent, slug: 'one', width: WIDGET_WIDTHS.QUARTER },
			{ component: WidgetComponent, slug: 'two', width: WIDGET_WIDTHS.QUARTER },
			{ component: WidgetComponent, slug: 'three', width: WIDGET_WIDTHS.QUARTER },
			{ component: WidgetComponent, slug: 'four', width: WIDGET_WIDTHS.FULL },
		] );
		container = render( <WidgetAreaRenderer slug="gridcell-test" />, { registry } ).container;
		expect( container.firstChild.querySelectorAll( '.googlesitekit-widget-area-widgets' )[ 0 ] ).toMatchSnapshot();

		registry = createTestRegistryWithArea( 'gridcell-test' );
		createWidgets( registry, [
			{ component: WidgetComponent, slug: 'one', width: WIDGET_WIDTHS.QUARTER },
			{ component: WidgetComponent, slug: 'two', width: WIDGET_WIDTHS.QUARTER },
			{ component: WidgetComponent, slug: 'three', width: WIDGET_WIDTHS.QUARTER },
			{ component: WidgetComponent, slug: 'four', width: WIDGET_WIDTHS.FULL },
			{ component: WidgetComponent, slug: 'five', width: WIDGET_WIDTHS.QUARTER },
			{ component: WidgetComponent, slug: 'six', width: WIDGET_WIDTHS.QUARTER },
			{ component: WidgetComponent, slug: 'seven', width: WIDGET_WIDTHS.QUARTER },
		] );
		container = render( <WidgetAreaRenderer slug="gridcell-test" />, { registry } ).container;
		expect( container.firstChild.querySelectorAll( '.googlesitekit-widget-area-widgets' )[ 0 ] ).toMatchSnapshot();
	} );

	it( 'should treat widgets that render no content as zero-width (ignoring them)', () => {
		registry = createTestRegistryWithArea( 'gridcell-test' );
		createWidgets( registry, [
			{ component: WidgetComponent, slug: 'one', width: WIDGET_WIDTHS.QUARTER },
			{ component: WidgetComponentEmpty, slug: 'empty', width: WIDGET_WIDTHS.HALF },
			{ component: WidgetComponent, slug: 'three', width: WIDGET_WIDTHS.QUARTER },
		] );

		// Used to mute the forwardRef error React will throw here.
		muteConsole( 'error', 1 );
		const { container } = render( <WidgetAreaRenderer slug="gridcell-test" />, { registry } );

		expect( container.firstChild.querySelectorAll( '.googlesitekit-widget-area-widgets' )[ 0 ] ).toMatchSnapshot();
	} );

	it( 'should not resize widgets that fit into a 12-column grid', () => {
		registry = createTestRegistryWithArea( 'gridcell-test' );
		createWidgets( registry, [
			{ component: WidgetComponent, slug: 'one', width: WIDGET_WIDTHS.QUARTER },
			{ component: WidgetComponent, slug: 'two', width: WIDGET_WIDTHS.QUARTER },
			{ component: WidgetComponent, slug: 'three', width: WIDGET_WIDTHS.QUARTER },
			{ component: WidgetComponent, slug: 'four', width: WIDGET_WIDTHS.QUARTER },
		] );

		let container = render( <WidgetAreaRenderer slug="gridcell-test" />, { registry } ).container;
		expect( container.firstChild.querySelectorAll( '.googlesitekit-widget-area-widgets' )[ 0 ] ).toMatchSnapshot();

		registry = createTestRegistryWithArea( 'gridcell-test' );
		createWidgets( registry, [
			{ component: WidgetComponent, slug: 'one', width: WIDGET_WIDTHS.HALF },
			{ component: WidgetComponent, slug: 'two', width: WIDGET_WIDTHS.HALF },
		] );
		container = render( <WidgetAreaRenderer slug="gridcell-test" />, { registry } ).container;
		expect( container.firstChild.querySelectorAll( '.googlesitekit-widget-area-widgets' )[ 0 ] ).toMatchSnapshot();

		registry = createTestRegistryWithArea( 'gridcell-test' );
		createWidgets( registry, [
			{ component: WidgetComponent, slug: 'one', width: WIDGET_WIDTHS.QUARTER },
			{ component: WidgetComponent, slug: 'two', width: WIDGET_WIDTHS.HALF },
			{ component: WidgetComponent, slug: 'three', width: WIDGET_WIDTHS.QUARTER },
		] );
		container = render( <WidgetAreaRenderer slug="gridcell-test" />, { registry } ).container;
		expect( container.firstChild.querySelectorAll( '.googlesitekit-widget-area-widgets' )[ 0 ] ).toMatchSnapshot();
	} );
} );
