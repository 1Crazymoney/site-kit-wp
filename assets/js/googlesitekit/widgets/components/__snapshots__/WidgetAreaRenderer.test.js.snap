// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WidgetAreaRenderer should resize widgets so they fill a row 1`] = `
<div
  class="googlesitekit-widget-area-widgets"
>
  <div
    class="mdc-layout-grid__inner"
  >
    <div
      class="googlesitekit-widget googlesitekit-widget--one mdc-layout-grid__cell mdc-layout-grid__cell--span-12"
    >
      <div>
        Foo bar!
      </div>
    </div>
    <div
      class="googlesitekit-widget googlesitekit-widget--two mdc-layout-grid__cell mdc-layout-grid__cell--span-4-desktop mdc-layout-grid__cell--span-4-tablet"
    >
      <div>
        Foo bar!
      </div>
    </div>
    <div
      class="googlesitekit-widget googlesitekit-widget--three mdc-layout-grid__cell mdc-layout-grid__cell--span-8-desktop mdc-layout-grid__cell--span-8-tablet"
    >
      <div>
        Foo bar!
      </div>
    </div>
  </div>
</div>
`;

exports[`WidgetAreaRenderer should resize widgets so they fill a row 2`] = `
<div
  class="googlesitekit-widget-area-widgets"
>
  <div
    class="mdc-layout-grid__inner"
  >
    <div
      class="googlesitekit-widget googlesitekit-widget--one mdc-layout-grid__cell mdc-layout-grid__cell--span-3-desktop mdc-layout-grid__cell--span-4-tablet"
    >
      <div>
        Foo bar!
      </div>
    </div>
    <div
      class="googlesitekit-widget googlesitekit-widget--two mdc-layout-grid__cell mdc-layout-grid__cell--span-12"
    >
      <div>
        Foo bar!
      </div>
    </div>
    <div
      class="googlesitekit-widget googlesitekit-widget--three mdc-layout-grid__cell mdc-layout-grid__cell--span-3-desktop mdc-layout-grid__cell--span-4-tablet"
    >
      <div>
        Foo bar!
      </div>
    </div>
    <div
      class="googlesitekit-widget googlesitekit-widget--four mdc-layout-grid__cell mdc-layout-grid__cell--span-3-desktop mdc-layout-grid__cell--span-4-tablet"
    >
      <div>
        Foo bar!
      </div>
    </div>
  </div>
</div>
`;

exports[`WidgetAreaRenderer should resize widgets so they fill a row 3`] = `
<div
  class="googlesitekit-widget-area-widgets"
>
  <div
    class="mdc-layout-grid__inner"
  >
    <div
      class="googlesitekit-widget googlesitekit-widget--one mdc-layout-grid__cell mdc-layout-grid__cell--span-4-desktop mdc-layout-grid__cell--span-4-tablet"
    >
      <div>
        Foo bar!
      </div>
    </div>
    <div
      class="googlesitekit-widget googlesitekit-widget--two mdc-layout-grid__cell mdc-layout-grid__cell--span-12"
    >
      <div>
        Foo bar!
      </div>
    </div>
    <div
      class="googlesitekit-widget googlesitekit-widget--three mdc-layout-grid__cell mdc-layout-grid__cell--span-4-desktop mdc-layout-grid__cell--span-4-tablet"
    >
      <div>
        Foo bar!
      </div>
    </div>
    <div
      class="googlesitekit-widget googlesitekit-widget--four mdc-layout-grid__cell mdc-layout-grid__cell--span-8-desktop mdc-layout-grid__cell--span-8-tablet"
    >
      <div>
        Foo bar!
      </div>
    </div>
    <div
      class="googlesitekit-widget googlesitekit-widget--five mdc-layout-grid__cell mdc-layout-grid__cell--span-8-desktop mdc-layout-grid__cell--span-8-tablet"
    >
      <div>
        Foo bar!
      </div>
    </div>
  </div>
</div>
`;

exports[`WidgetAreaRenderer should resize widgets so they fill a row 4`] = `
<div
  class="googlesitekit-widget-area-widgets"
>
  <div
    class="mdc-layout-grid__inner"
  >
    <div
      class="googlesitekit-widget googlesitekit-widget--one mdc-layout-grid__cell mdc-layout-grid__cell--span-3-desktop mdc-layout-grid__cell--span-4-tablet"
    >
      <div>
        Foo bar!
      </div>
    </div>
    <div
      class="googlesitekit-widget googlesitekit-widget--two mdc-layout-grid__cell mdc-layout-grid__cell--span-3-desktop mdc-layout-grid__cell--span-4-tablet"
    >
      <div>
        Foo bar!
      </div>
    </div>
    <div
      class="googlesitekit-widget googlesitekit-widget--three mdc-layout-grid__cell mdc-layout-grid__cell--span-6-desktop mdc-layout-grid__cell--span-8-tablet"
    >
      <div>
        Foo bar!
      </div>
    </div>
  </div>
</div>
`;

exports[`WidgetAreaRenderer should resize widgets so they fill a row 5`] = `
<div
  class="googlesitekit-widget-area-widgets"
>
  <div
    class="mdc-layout-grid__inner"
  >
    <div
      class="googlesitekit-widget googlesitekit-widget--one mdc-layout-grid__cell mdc-layout-grid__cell--span-3-desktop mdc-layout-grid__cell--span-4-tablet"
    >
      <div>
        Foo bar!
      </div>
    </div>
    <div
      class="googlesitekit-widget googlesitekit-widget--two mdc-layout-grid__cell mdc-layout-grid__cell--span-3-desktop mdc-layout-grid__cell--span-4-tablet"
    >
      <div>
        Foo bar!
      </div>
    </div>
    <div
      class="googlesitekit-widget googlesitekit-widget--three mdc-layout-grid__cell mdc-layout-grid__cell--span-4-desktop mdc-layout-grid__cell--span-4-tablet"
    >
      <div>
        Foo bar!
      </div>
    </div>
    <div
      class="googlesitekit-widget googlesitekit-widget--four mdc-layout-grid__cell mdc-layout-grid__cell--span-8-desktop mdc-layout-grid__cell--span-8-tablet"
    >
      <div>
        Foo bar!
      </div>
    </div>
  </div>
</div>
`;

exports[`WidgetAreaRenderer should resize widgets so they fill a row 6`] = `
<div
  class="googlesitekit-widget-area-widgets"
>
  <div
    class="mdc-layout-grid__inner"
  >
    <div
      class="googlesitekit-widget googlesitekit-widget--one mdc-layout-grid__cell mdc-layout-grid__cell--span-4-desktop mdc-layout-grid__cell--span-4-tablet"
    >
      <div>
        Foo bar!
      </div>
    </div>
    <div
      class="googlesitekit-widget googlesitekit-widget--two mdc-layout-grid__cell mdc-layout-grid__cell--span-4-desktop mdc-layout-grid__cell--span-4-tablet"
    >
      <div>
        Foo bar!
      </div>
    </div>
    <div
      class="googlesitekit-widget googlesitekit-widget--three mdc-layout-grid__cell mdc-layout-grid__cell--span-4-desktop mdc-layout-grid__cell--span-4-tablet"
    >
      <div>
        Foo bar!
      </div>
    </div>
    <div
      class="googlesitekit-widget googlesitekit-widget--four mdc-layout-grid__cell mdc-layout-grid__cell--span-12"
    >
      <div>
        Foo bar!
      </div>
    </div>
  </div>
</div>
`;

exports[`WidgetAreaRenderer should resize widgets so they fill a row 7`] = `
<div
  class="googlesitekit-widget-area-widgets"
>
  <div
    class="mdc-layout-grid__inner"
  >
    <div
      class="googlesitekit-widget googlesitekit-widget--one mdc-layout-grid__cell mdc-layout-grid__cell--span-4-desktop mdc-layout-grid__cell--span-4-tablet"
    >
      <div>
        Foo bar!
      </div>
    </div>
    <div
      class="googlesitekit-widget googlesitekit-widget--two mdc-layout-grid__cell mdc-layout-grid__cell--span-4-desktop mdc-layout-grid__cell--span-4-tablet"
    >
      <div>
        Foo bar!
      </div>
    </div>
    <div
      class="googlesitekit-widget googlesitekit-widget--three mdc-layout-grid__cell mdc-layout-grid__cell--span-4-desktop mdc-layout-grid__cell--span-4-tablet"
    >
      <div>
        Foo bar!
      </div>
    </div>
    <div
      class="googlesitekit-widget googlesitekit-widget--four mdc-layout-grid__cell mdc-layout-grid__cell--span-12"
    >
      <div>
        Foo bar!
      </div>
    </div>
    <div
      class="googlesitekit-widget googlesitekit-widget--five mdc-layout-grid__cell mdc-layout-grid__cell--span-4-desktop mdc-layout-grid__cell--span-4-tablet"
    >
      <div>
        Foo bar!
      </div>
    </div>
    <div
      class="googlesitekit-widget googlesitekit-widget--six mdc-layout-grid__cell mdc-layout-grid__cell--span-4-desktop mdc-layout-grid__cell--span-4-tablet"
    >
      <div>
        Foo bar!
      </div>
    </div>
    <div
      class="googlesitekit-widget googlesitekit-widget--seven mdc-layout-grid__cell mdc-layout-grid__cell--span-4-desktop mdc-layout-grid__cell--span-4-tablet"
    >
      <div>
        Foo bar!
      </div>
    </div>
  </div>
</div>
`;

exports[`WidgetAreaRenderer should return the same number of elements as widgets from a selector 1`] = `
<div
  class="googlesitekit-widget-area-widgets"
>
  <div
    class="mdc-layout-grid__inner"
  >
    <div
      class="googlesitekit-widget googlesitekit-widget--one mdc-layout-grid__cell mdc-layout-grid__cell--span-12"
    >
      <div>
        Foo bar!
      </div>
    </div>
    <div
      class="googlesitekit-widget googlesitekit-widget--two mdc-layout-grid__cell mdc-layout-grid__cell--span-12"
    >
      <div>
        Foo bar!
      </div>
    </div>
    <div
      class="googlesitekit-widget googlesitekit-widget--three mdc-layout-grid__cell mdc-layout-grid__cell--span-12"
    >
      <div>
        Foo bar!
      </div>
    </div>
  </div>
</div>
`;

exports[`WidgetAreaRenderer should treat widgets that render no content as zero-width (ignoring them) 1`] = `
<div
  class="googlesitekit-widget-area-widgets"
>
  <div
    class="mdc-layout-grid__inner"
  >
    <div
      class="googlesitekit-widget googlesitekit-widget--one mdc-layout-grid__cell mdc-layout-grid__cell--span-3-desktop mdc-layout-grid__cell--span-4-tablet"
    >
      <div>
        Foo bar!
      </div>
    </div>
    <div
      class="googlesitekit-widget googlesitekit-widget--empty googlesitekit-widget-area--hidden"
    />
    <div
      class="googlesitekit-widget googlesitekit-widget--three mdc-layout-grid__cell mdc-layout-grid__cell--span-3-desktop mdc-layout-grid__cell--span-4-tablet"
    >
      <div>
        Foo bar!
      </div>
    </div>
  </div>
</div>
`;
